platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
You'll still have the same bunch of source files and Platform classes (though it's better to use different names and typedef used one to Platform) - there's no need to use run-time polymorphism here as there's only a single Platform implementation at a time. But virtual class will give you have strictly defined interface (you can forget to fill a variable in current implementation; you can't forget to implement virtual method) and an ability to keep common code in intermediate classes. My idea is

```
// platform.h
class IPlatform {
   virtual string GetExecutableName() = 0;
   virtual string GetSettingsDir() = 0;
};
```
```
// platform_unix.h
class UnixPlatform : public IPlatform {
    virtual string GetSettingsDir() override { return getenv("HOME") + "./mapsme"; }
};
```
```
// platform_freebsd.h
class FreeBSDPlatform : public UnixPlatform {
    virtual string GetExecutablePath() final { return ... }
};

typedef FreeBSDPlatform Platform; // all consumers work with Platform -
                                  // a final class, no runtime polymorphism
```